.github/workflows/ci.yml

```diff
@@ -0,0 +1,85 @@
+name: Continuous Integration
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches:
+      - "**"
+  workflow_dispatch:
+    branches:
+      - "**"
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+env:
+    CARGO_TERM_COLOR: always
+
+jobs:
+  test:
+    name: Test Suite
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    if: "! contains(toJSON(github.event.commits.*.message), '[skip-ci]')"
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v4
+
+      - name: Install nightly toolchain
+        uses: dtolnay/rust-toolchain@nightly
+
+      - name: Set up rust cache
+        uses: Swatinem/rust-cache@v2
+        with:
+            cache-on-failure: true
+
+      - name: Check in protocol_decoder subdirectory
+        run: cargo check --manifest-path protocol_decoder/Cargo.toml
+        env:
+          RUSTFLAGS: -Copt-level=3 -Cdebug-assertions -Coverflow-checks=y -Cdebuginfo=0
+          RUST_LOG: 1
+          CARGO_INCREMENTAL: 1
+          RUST_BACKTRACE: 1
```

Comment:
I'm not super familiar with GitHub's CI, but I wonder if there's a nice way to avoid duplicating the `env` variables for these tasks? I know we could define `env` one "layer" above and it would be present for all of the tasks, but then it means these variables would be defined for tasks which do not need them. Not obvious what to do here after looking this up myself, but yeah. Fine to leave this.

Reply:
I'll have to admit I'm not an expert in setting CI jobs, I basically got inspiration from the latest plonky2 (got refactored yesterday).